From 8d76b59f3f158b49f8815784745e5c2f0fafbb52 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@lispysnake.com>
Date: Fri, 13 Mar 2020 11:34:17 +0000
Subject: [PATCH] feature: Fix compilation of GTK with newer GLib

If the test builds don't pass, we can't compile perf with GTK2 due to
upstream GLib marking GTypeDebugFlags/GTimeVal, etc, as deprecated.
Thus, skip the deprecation warning, and allow feature detection to
work.

Signed-off-by: Ikey Doherty <ikey.doherty@lispysnake.com>
---
 tools/build/feature/test-gtk2-infobar.c | 1 +
 tools/build/feature/test-gtk2.c         | 1 +
 2 files changed, 2 insertions(+)

diff --git a/tools/build/feature/test-gtk2-infobar.c b/tools/build/feature/test-gtk2-infobar.c
index b1b716dd5..746946709 100644
--- a/tools/build/feature/test-gtk2-infobar.c
+++ b/tools/build/feature/test-gtk2-infobar.c
@@ -1,5 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
 #pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #include <gtk/gtk.h>
 #pragma GCC diagnostic error "-Wstrict-prototypes"
 
diff --git a/tools/build/feature/test-gtk2.c b/tools/build/feature/test-gtk2.c
index 2aaf4bfc2..8a1bb8ca3 100644
--- a/tools/build/feature/test-gtk2.c
+++ b/tools/build/feature/test-gtk2.c
@@ -1,5 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
 #pragma GCC diagnostic ignored "-Wstrict-prototypes"
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #include <gtk/gtk.h>
 #pragma GCC diagnostic error "-Wstrict-prototypes"
 
-- 
2.25.1

